ns_api.py

```diff
@@ -445,8 +444,10 @@ def __init__(self, part_dict=None):
         self.has_delay = False
         if part_dict['cancelled']:
             self.going = False
-        if 'punctuality' in part_dict and part_dict['punctuality'] != 100.0:
-            self.has_delay = True
+        try:
+            self.punctuality = part_dict['punctuality']
```

Comment:
This could even be simplified with a part_dict.get('punctionality', None) which means it will fall back to None if it can't find a value for that key.